src/main/java/gregtech/GT_Mod.java

```diff
@@ -60,7 +60,8 @@
 
 @Mod(modid = "gregtech", name = "GregTech", version = "MC1710", useMetadata = false, dependencies = "required-after:IC2; after:Forestry; after:PFAAGeologica; after:Thaumcraft; after:Railcraft; after:appliedenergistics2; after:ThermalExpansion; after:TwilightForest; after:harvestcraft; after:magicalcrops; after:BuildCraft|Transport; after:BuildCraft|Silicon; after:BuildCraft|Factory; after:BuildCraft|Energy; after:BuildCraft|Core; after:BuildCraft|Builders; after:GalacticraftCore; after:GalacticraftMars; after:GalacticraftPlanets; after:ThermalExpansion|Transport; after:ThermalExpansion|Energy; after:ThermalExpansion|Factory; after:RedPowerCore; after:RedPowerBase; after:RedPowerMachine; after:RedPowerCompat; after:RedPowerWiring; after:RedPowerLogic; after:RedPowerLighting; after:RedPowerWorld; after:RedPowerControl; after:UndergroundBiomes;")
 public class GT_Mod implements IGT_Mod {
-    public static final int VERSION = 509;
+    public static final int VERSION = 509, SUBVERSION = 31;
+    public static final int TOTAL_VERSION = 1000 * VERSION + SUBVERSION;

```

Comment:
An example:
I'm a developer of a new addon for GT. I don't care how this version should be calculated I just want compare the current version and 35 versions. Then I can write this
```
if(GT_Mod.TOTAL_VERSION > GT_Utils.calculateVersion(35) {
```
Sent the only one argument cuz 509 can be default so in real is two methods
`calculateVersion(int majorVersion, int minorVersion)`
`calculateVersion(int minorVersion) //cals calculateVersion(GT_Mod.VERSION, minorVersion)`
Also the magic numbers in the code is an antipattern.